core/architecture.md

```diff
@@ -0,0 +1,82 @@
+# Nimbess Controller Architecture
+
+## Overview
+
+Nimbess is a light-weight distributed controller architecture that runs as a
+Kubernetes pod across every Kubelet node. Nimbess by default supports the
+Berkley Extensible Software Switch [BESS](https://github.com/NetSys/bess)
+dataplane in order to achieve maximum network pipeline programmability and
+performance. The primary goal of Nimbess is to change the approach to the
+networking paradigm in Kubernetes itself by implementing Unified Network Policy
+across Kubernetes. More on how this works can be found in the 
+[Functional Design Guide](functional-spec.md). The purpose of this document is
+to describe the software architecture and design.
+
+## Background
+
+Typical CNI or networking plugin designs for Kubernetes either rely on an SDN
+controller/dataplane combination and/or multiple external applications such
+as iptables, haproxy/envoy, conntrack, etc in order to achieve a fully featured
+network dataplane within Kubernetes. These features include services, ingress,
+and also "network policy". The Network Policy resource in Kubernetes is limited
+and refers only to access-control policy of "who can talk to who". 
+
+One of the main goals of Nimbess is to simplify and consolidate these features
+into a single point of entry using a custom resource definition (CRD) called
+Unified Network Policy (UNP). The UNP is capable of not only implementing
+access-control lists (ACL), but also network slicing by including
+multi-tenancy, bandwidth requirements, traffic differentiation (QoS), 
+Service Function Chaining (SFC), etc.
+
+Additionally, CNI implementations are also inherently limited in Kubernetes,
+due the nature of the specification which limits CNI invocation to pod create
+or delete calls. Efforts have been made to work around this issue using Network
+Service Mesh ([NSM](https://github.com/networkservicemesh)), which allows for
+dynamic update of ports and network fabric throughout the pod lifecycle.
+Furthermore, the CNI spec only allows by default for a single port to be
+created. Efforts like the [Multus](https://github.com/intel/multus-cni)
+project have been implemented to work around this issue by allowing multiple
+CNIs, and thus multiple ports to be created.
+
+Nimbess resolves both of these issues in different ways. By default, when a pod
+is created, Nimbess places a single port for that pod in a default network. It
```

Comment:
change "in a default network" to "with default connectivity"

Reply:
yeah and need to update this to our 2 port (1 kernel, 1 dpdk) current proposal (may change again in the future)